io/include/gz/common/CSVStreams.hh

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright (C) 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+#ifndef GZ_COMMON_CSVSTREAMS_HH_
+#define GZ_COMMON_CSVSTREAMS_HH_
+
+#include <istream>
+#include <string>
+#include <vector>
+
+#include <gz/common/io/Export.hh>
+
+#include <gz/utils/ImplPtr.hh>
+
+namespace gz
+{
+  namespace common
+  {
+    /// \brief A CSV specification.
+    struct CSVDialect
+    {
+      /// Field delimiter character.
+      char delimiter;
+
+      /// Row termination character.
+      char terminator;
+
+      /// Field quoting character.
+      char quote;
+
+      /// CSV dialect as expected by Unix tools.
+      static const CSVDialect Unix;
```

Comment:
Does this parser work with Windows for example? If not I think we should mention it somewhere.

Reply:
@caguero I'll start by saying I didn't try this on Windows, BUT my understanding is that Windows will translate `\r\n` to `\n` when reading a text file (see [here](https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/fopen-wfopen?view=msvc-170#generic-text-routine-mappings)) so this dialect should work just fine there too.